packages/react-dom/src/__tests__/ReactInputSelection-test.js

```diff
@@ -0,0 +1,254 @@
+/**
+ * Copyright (c) 2013-present, Facebook, Inc.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @emails react-core
+ */
+
+'use strict';
+
+var React = require('react');
+var ReactDOM = require('react-dom');
+var ReactTestUtils = require('react-dom/test-utils');
+var ReactInputSelection = require('../client/ReactInputSelection');
+
+describe('ReactInputSelection', () => {
+  var textValue = 'the text contents';
+  var createAndMountElement = (type, props, children) => {
+    var element = React.createElement(type, props, children);
+    var instance = ReactTestUtils.renderIntoDocument(element);
+    return ReactDOM.findDOMNode(instance);
+  };
+  var makeGetSelection = (win = window) => () => ({
+    anchorNode: win.document.activeElement,
+    focusNode: win.document.activeElement,
+    anchorOffset:
+      win.document.activeElement && win.document.activeElement.selectionStart,
+    focusOffset:
+      win.document.activeElement && win.document.activeElement.selectionEnd,
+  });
+
+  describe('hasSelectionCapabilities', () => {
+    it('returns true for textareas', () => {
+      var textarea = document.createElement('textarea');
+      expect(ReactInputSelection.hasSelectionCapabilities(textarea)).toBe(true);
+    });
+
+    it('returns true for text inputs', () => {
+      var inputText = document.createElement('input');
+      var inputReadOnly = document.createElement('input');
+      inputReadOnly.readOnly = 'true';
+      var inputNumber = document.createElement('input');
+      inputNumber.type = 'number';
+      var inputEmail = document.createElement('input');
+      inputEmail.type = 'email';
+      var inputPassword = document.createElement('input');
+      inputPassword.type = 'password';
+      var inputHidden = document.createElement('input');
+      inputHidden.type = 'hidden';
+
+      expect(ReactInputSelection.hasSelectionCapabilities(inputText)).toBe(
+        true,
+      );
+      expect(ReactInputSelection.hasSelectionCapabilities(inputReadOnly)).toBe(
+        true,
+      );
+      expect(ReactInputSelection.hasSelectionCapabilities(inputNumber)).toBe(
+        false,
+      );
+      expect(ReactInputSelection.hasSelectionCapabilities(inputEmail)).toBe(
+        false,
+      );
+      expect(ReactInputSelection.hasSelectionCapabilities(inputPassword)).toBe(
+        false,
+      );
+      expect(ReactInputSelection.hasSelectionCapabilities(inputHidden)).toBe(
+        false,
+      );
+    });
+
+    it('returns true for contentEditable elements', () => {
+      var div = document.createElement('div');
+      div.contentEditable = 'true';
+      var body = document.createElement('body');
+      body.contentEditable = 'true';
+      var input = document.createElement('input');
+      input.contentEditable = 'true';
```

Comment:
Huh. I didn't know you could do this.

Reply:
<img width="438" alt="screen shot 2017-11-16 at 4 39 22 pm" src="https://user-images.githubusercontent.com/590904/32917297-ba11e4f4-caec-11e7-8e0b-cf010973cb5b.png">

How fun.